src/lib/TopLevel.hs

```diff
@@ -624,12 +632,23 @@ execUDecl mname decl = do
     UDeclResultDone sourceMap' -> emitSourceMap sourceMap'
 {-# SCC execUDecl #-}
 
+lowerTopLevelFun :: (Topper m, Mut n) => AtomName n -> m n ()
+lowerTopLevelFun fname = do
+  fPreSimp <- specializedFunPreSimpDefinition fname
+  fSimp <- simplifyTopFunction fPreSimp
+  evalRequiredSpecializations fSimp
+  fSimp' <- hoistIxDicts fSimp
+  -- TODO: actually do the lowering (lowering only handles blocks, not functions, right now)
```

Comment:
So if we plug in the destination passing etc. here, then it will also apply to Ix dicts?

Reply:
Yes, if we want it to (or we can make it take a different path if we prefer).